File: frontend/src/types/projectOverview.ts
import type { AISummary } from "./aiSummary";
import type { Bug } from "./bug";
import type { DailyNote } from "./dailyNote";
import type { Decision } from "./decision";
import type { ProjectDocument } from "./document";
import type { Project } from "./project";

export interface ProjectOverviewCounts {
  documents_count: number;
  notes_count: number;
  bugs_count: number;
  open_bugs_count: number;
  decisions_count: number;
  summaries_count: number;
}

export interface ProjectMemoryCoverage {
  has_documents: boolean;
  has_notes: boolean;
  has_bugs: boolean;
  has_decisions: boolean;
  has_ai_summary: boolean;
}

export interface ProjectOverview {
  project: Project;
  counts: ProjectOverviewCounts;
  coverage: ProjectMemoryCoverage;
  latest_documents: ProjectDocument[];
  latest_notes: DailyNote[];
  latest_bugs: Bug[];
  latest_decisions: Decision[];
  latest_summary: AISummary | null;
  generated_at: string;
}

File: backend/app/schemas/project_overview.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel

from app.schemas.ai_summary import AISummaryRead
from app.schemas.bug import BugRead
from app.schemas.daily_note import DailyNoteRead
from app.schemas.decision import DecisionRead
from app.schemas.document import DocumentRead
from app.schemas.project import ProjectRead


class ProjectOverviewCounts(BaseModel):
    documents_count: int
    notes_count: int
    bugs_count: int
    open_bugs_count: int
    decisions_count: int
    summaries_count: int


class ProjectMemoryCoverage(BaseModel):
    has_documents: bool
    has_notes: bool
    has_bugs: bool
    has_decisions: bool
    has_ai_summary: bool


class ProjectOverviewResponse(BaseModel):
    project: ProjectRead
    counts: ProjectOverviewCounts
    coverage: ProjectMemoryCoverage
    latest_documents: list[DocumentRead]
    latest_notes: list[DailyNoteRead]
    latest_bugs: list[BugRead]
    latest_decisions: list[DecisionRead]
    latest_summary: Optional[AISummaryRead] = None
    generated_at: datetime
